src/db/tests/batch.rs

```diff
@@ -242,3 +246,50 @@ async fn quota_test_append_batch() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test_append_async_w_null() -> Result<()> {
+    let settings = crate::settings::test_settings();
+    if !settings.uses_spanner() {
+        dbg!("### Skipping test for mysql");
+        return Ok(());
+    }
+    let pool = db_pool(Some(settings)).await?;
+    let db = test_db(pool.as_ref()).await?;
+    let ttl = crate::db::util::ms_since_epoch() as u32;
+
+    let uid = 1;
+    let coll = "clients";
+    let first_bsos = vec![postbso(
+        "b0",
+        Some("payload 0"),
+        Some(10),
+        Some(ttl + 10_000),
+    )];
+    let new_batch = db.create_batch(cb(uid, coll, first_bsos)).await?;
+
+    // update the single bso twice, once changing
+    let updated_bsos = vec![
+        postbso("b0", None, Some(10), None),
+        postbso("b1", Some("payload 3"), None, Some(ttl + 20_000)),
+    ];
+
+    db.append_to_batch(ab(uid, coll, new_batch.clone(), updated_bsos))
+        .await?;
+
+    assert!(db
+        .get_batch(gb(uid, coll, new_batch.id.clone()))
+        .await?
+        .is_some());
+    // ideally, extract the updated batch_bso record out of SpannerDB and compare it against what we just
```

Comment:
I'd modify / add to this test a couple things (I don't think it's *too* much extra code): 

call `append_to_batch` with updates to "b0" twice. Maybe ttl the first time, sortindex the second.

Then copy some of the code from something like `append_commit` above: call `commit_batch`, then `get_bso(.. "b0")`. Finally assert payload is still "payload 0", but ttl and sortindex were updated

Reply:
Crap, yeah, I had dropped the second call because of the mysql thing. Let me add it back in.

Sadly, looks like I'll also need to land #853 first because of protobuf fun.
Thankfully, that didn't impact any code.